lib/experimental/fonts/font-library/class-wp-rest-font-collections-controller.php

```diff
@@ -0,0 +1,283 @@
+ *
+ * @since 6.4.0
+ */
+class WP_REST_Font_Collections_Controller extends WP_REST_Controller {
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 6.4.0
+	 */
+	public function __construct() {
+		$this->rest_base = 'font-collections';
+		$this->namespace = 'wp/v2';
+	}
+
+	/**
+	 * Registers the routes for the objects of the controller.
+	 *
+	 * @since 6.4.0
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'update_font_library_permissions_check' ),
+				),
+				'schema' => array( $this, 'get_items_schema' ),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<id>[\/\w-]+)',
+			array(
+				'args'   => array(
+					'id' => array(
+						'description' => __( 'Unique identifier for the post.' ),
+						'type'        => 'string',
+						'required'    => true,
+					),
+				),
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => array( $this, 'update_font_library_permissions_check' ),
+				),
+				'schema' => array( $this, 'get_item_schema' ),
+			)
+		);
+	}
+
+	/**
+	 * Gets a font collection.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_item( $request ) {
+		$id         = $request['id'];
+		$collection = WP_Font_Library::get_font_collection( $id );
+		// If the collection doesn't exist returns a 404.
+		if ( is_wp_error( $collection ) ) {
+			$collection->add_data( array( 'status' => 404 ) );
+			return $collection;
+		}
+		$collection_with_data = $collection->get_data();
+		// If there was an error getting the collection data, return the error.
+		if ( is_wp_error( $collection_with_data ) ) {
+			$collection_with_data->add_data( array( 'status' => 500 ) );
+			return $collection_with_data;
+		}
+
+		return rest_ensure_response( $collection_with_data );
+	}
+
+	/**
+	 * Gets the font collections available.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response Response object.
+	 */
+	public function get_items( $request ) {
+		$collections = array();
+		foreach ( WP_Font_Library::get_font_collections() as $collection ) {
+			$collections[] = $collection->get_config();
+		}
+		return rest_ensure_response( $collections );
+	}
+
+	/**
+	 * Checks whether the user has permissions to update the Font Library.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @return true|WP_Error True if the request has write access for the item, WP_Error object otherwise.
+	 */
+	public function update_font_library_permissions_check() {
```

Comment:
If we only have `WP_REST_Server::READABLE` endpoints here, this should be `get_items_permissions_check()`, e.g.

https://github.com/WordPress/wordpress-develop/blob/6528d9840b6800edfb8e75c4924936bb14b3f31b/src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php#L146-L168